util.js

```diff
@@ -30,14 +33,14 @@ function isInvalidAsyncPlugin (plugin) {
   return plugin.length !== 2 && plugin.constructor.name === 'AsyncFunction'
 }
 
-function requireServerPluginFromPath (modulePath) {
+async function requireServerPluginFromPath (modulePath) {
   const resolvedModulePath = path.resolve(process.cwd(), modulePath)
 
   if (!fs.existsSync(resolvedModulePath)) {
     throw new Error(`${resolvedModulePath} doesn't exist within ${process.cwd()}`)
   }
 
-  const serverPlugin = require(resolvedModulePath)
+  const serverPlugin = hasESM ? (await import(url.pathToFileURL(resolvedModulePath).href)).default : require(resolvedModulePath)
```

Comment:
I would prefer not to load cjs via esm.

Reply:
I tried the avoid the complexity present in fastify-autoload of having to redo nodes internal algorithm to determine what is CJS and ESM. If in a years time the node community changes the internal algorithm we need to change this code as well. That is what I was trying to avoid.

The [NodeJS documentation](https://nodejs.org/api/esm.html#esm_import_expressions) shows that 'import()' is allowed to be used to import CJS.
```js
import * as m from 'cjs';
console.log(m);
console.log(m === await import('cjs'));
// Prints:
//   [Module] { default: <module.exports> }
//   true
```
Do you still want the the additional complexity ?

Reply:
Yes. cjs over esm go through some other code branch.